src/carbonado.rs

```diff
@@ -321,28 +321,33 @@ pub async fn retrieve(
 
 #[cfg(feature = "server")]
 pub async fn handle_file(pk: &str, name: &str, bytes: usize) -> Result<PathBuf, CarbonadoError> {
-    use percent_encoding::percent_decode;
+    use crate::carbonado::constants::FORM;
+    use percent_encoding::{percent_decode_str, utf8_percent_encode};
 
     let network = NETWORK.read().await.to_string();
     let mut final_name = name.to_string();
     if !name.contains(&network) {
         final_name = format!("{network}/{name}");
     }
 
-    let directory = std::path::Path::new(
+    let reencoded = utf8_percent_encode(&percent_decode_str(name).decode_utf8()?, FORM).to_string();
+    if name != reencoded {
```

Comment:
I'm not sure if it works.

Because to pass the file path, we need to run the url encode.

Ex:

This path params `filename` works because we encode the "`/`" to "`V%2F`"

http://.../carbonado/02...e/regtest/**memphis_asia_crash`V%2F`bitmask-fungible_assets_stock.c15**

But if we send without encode, the server understand the "/" is a path and return 404.

http://.../carbonado/02...e/regtest/**memphis_asia_crash`/`bitmask-fungible_assets_stock.c15**